driver-kotlin-coroutine/src/test/kotlin/com/mongodb/kotlin/client/coroutine/ListCollectionNamesFlowTest.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kotlin.client.coroutine
+
+import com.mongodb.reactivestreams.client.ListCollectionNamesPublisher
+import java.util.concurrent.TimeUnit
+import kotlin.reflect.full.declaredFunctions
+import kotlin.test.assertEquals
+import org.bson.BsonDocument
+import org.bson.BsonString
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.verifyNoMoreInteractions
+
+class ListCollectionNamesFlowTest {
```

Comment:
This test is based on `com.mongodb.kotlin.client.coroutine.ListCollectionsFlowTest`.